metrics/api/src/main/java/io/helidon/metrics/api/Wrapped.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.metrics.api;
+
+/**
+ * Behavior of a type that wraps a related type.
+ */
+public interface Wrapped {
+
+    /**
+     * Unwraps the wrapped item as the specified type.
+     *
+     * @param c {@link Class} to which to cast this object
+     * @return this object cast as the requested type
+     * @param <R> type to cast to
```

Comment:
I need to correct the JavaDoc in an important way; it's very sloppy as-is.

The concept with `Wrapped` is not that the `unwrap` method type-casts the instance but that each such interface actually has a differently-typed delegate that actually does the work.  `unwrap` is how a developer gets to the delegate with the correct type.

Trying to `unwrap` to a type that's not consistent with the delegate's actual type results in a `ClassCastException`.

In fact, the interface name should probably be `Wrapper` instead of `Wrapped` because it denotes the thing doing the wrapping, not the thing that is wrapped.

(I toyed with `Delegating` and `delegate()` but there is prior art in other areas toward the wrapping terminology so I stuck with that.)

Reply:
And all of that is fine; I was mainly looking for something like: does it return `null` if I pass a non-sensical value? Does it attempt to recursively unwrap? and so on. You get the idea.

Reply:
I've updated the JavaDoc to clarify.